cryosparc2/protocols/protocol_cryosparc_naive_local_refine.py

```diff
@@ -327,46 +308,26 @@ def createOutputStep(self):
         """
         Create the protocol output. Convert cryosparc file to Relion file
         """
-        import ast
         self._initializeUtilsVariables()
-        get_job_streamlog(self.projectName.get(), self.runLocalRefinement.get(),
-                          self._getFileName('stream_log'))
-
-        # Get the metadata information from stream.log
-        with open(self._getFileName('stream_log')) as f:
-            data = f.readlines()
-
-        x = ast.literal_eval(data[0])
-
-        # Find the ID of last iteration and the map resolution
-        for y in x:
-            if 'text' in y:
-                z = str(y['text'])
-                if z.startswith('FSC'):
-                    idd = y['imgfiles'][2]['fileid']
-                    itera = z[-3:]
-                elif 'Using Filter Radius' in z:
-                    nomRes = str(y['text']).split('(')[1].split(')')[0].replace(
-                        'A', 'Å')
-                    self.mapResolution = String(nomRes)
-                    self._store(self)
-                elif 'Estimated Bfactor' in z:
-                    estBFactor = str(y['text']).split(':')[1].replace('\n', '')
-                    self.estBFactor = String(estBFactor)
-                    self._store(self)
-
-        csParticlesName = ("cryosparc_" + self.projectName.get() + "_" +
-                           self.runLocalRefinement.get() + "_" + itera + "_particles.cs")
-        csFile = os.path.join(self.projectPath, self.projectName.get(),
-                              self.runLocalRefinement.get(), csParticlesName)
-
-        # Create the output folder
-        outputFolder = self._getExtraPath() + '/' + self.runLocalRefinement.get()
-        os.system("mkdir " + outputFolder)
-
-        # Copy the particles to scipion output folder
-        os.system("cp -r " + csFile + " " + outputFolder)
-        csFile = os.path.join(outputFolder, csParticlesName)
+        idd, itera = self.findLastIteration(self.runLocalRefinement.get())
```

Comment:
this part looks repeating in many protocols

Reply:
Many of the CS algorithms are executed by making a set of iterations which are recorded in a log file provided by CS called "stream_log". This file must be parsed (there is no other way) to determine the name of the output of the Last Iteration, the Estimated Bfactor, and other things(and not all of protocols generate the same structure file, even some with spelling mistakes that I have already reported).